### PR TITLE
Stats Manager - header only

### PR DESCRIPTION
**Pull request by rakhad**

### CODE REVIEW

File: Include/xsapi-c/stats_manager_c.h [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+{
+    /// <summary> A local user has been added </summary>
+    XblStatsManager_LocalUserAdded,
+
+    /// <summary> A local user has been removed </summary>
+    XblStatsManager_LocalUserRemoved,
+
+    /// <summary> A stat has been updated </summary>
+    XblStatsManager_StatUpdateComplete
+};
```
Comment (jplafonta) [inline Include/xsapi-c/stats_manager_c.h]: XblStatsManagerStatEventType

File: Include/xsapi-c/stats_manager_c.h [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    /// The stat value as a UTF-8 encoded string
+    /// </summary>
+    /// <remarks>Will be an empty string if dataType is not XblStatsManagerStatDataType_String</remarks>
+    _Field_z_ const char* asString;
+
+    /// <summary> 
+    /// Data type of the stat
+    /// </summary>
+    XblStatsManagerStatDataType dataType;
+};
```
Comment (jplafonta) [inline Include/xsapi-c/stats_manager_c.h]: XblStatValue (on below structs as well)

File: Include/xsapi-c/stats_manager_c.h [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+/// is getting suspended as this will try to update the stats even if it hasn't been 30 seconds 
+/// since the last flush. However requests to flush with isHighPriority=true are still limited 
+/// to a maximum rate of once every 30 seconds so it can't be used to flood the service.
+///
+/// Note that you may still be throttled when calling the stats service such as if you are 
+/// relaunching the title over and over and send stats at every launch.
+/// </remarks>
+STDAPI XblStatsManagerRequestFlushToService(
+    _In_ xbl_user_handle user,
+    _In_ bool isHighPriority = false
```
Comment (jplafonta) [inline Include/xsapi-c/stats_manager_c.h]: >= false [](start = 29, length = 7)

cannot have default param value in C

File: Include/xsapi-c/stats_manager_c.h [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+typedef struct XblStatEvent
+{   
+    /// <summary> 
+    /// Represents error code of the event
+    /// 0 if there is no error
+    /// </summary>
+    int32_t err;
+
+    /// <summary> 
+    /// Represents the error message of the event
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: >    /// Represents t [](start = 0, length = 20)

UTF-8 encoded

File: Include/xsapi-c/stats_manager_c.h [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+/// <summary> 
+/// Replaces the numerical stat by the value. Can be positive or negative.
+/// 
+/// Stats will be sent to the service automatically every 5 minutes or you can call 
+/// XblStatsManagerRequestFlushToService() if you want to update the service with 
+/// the latest stats sooner such as when a match or round ends.
+/// </summary>
+/// <param name="user">The local user whose stats to access.</param>
+/// <param name="statName">The name of the statistic to modify.</param>
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: >>The name of  [](start = 26, length = 13)

UTF-8 encoded

File: Include/xsapi-c/stats_manager_c.h [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: >">The name [](start = 25, length = 10)

UTF-8 encoded

File: Include/xsapi-c/stats_manager_c.h [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+/// <summary> 
+/// Replaces a string stat with the given value.
+/// 
+/// Stats will be sent to the service automatically every 5 minutes or you can call 
+/// XblStatsManagerRequestFlushToService() if you want to update the service with 
+/// the latest stats sooner such as when a match or round ends.
+/// </summary>
+/// <param name="user">The local user whose stats to access.</param>
+/// <param name="statName">The name of the statistic to modify.</param>
+/// <param name="statValue">The value to replace the stat by.</param>
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: UTF-8 encoded

File: Include/xsapi-c/stats_manager_c.h [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+    _In_ xbl_user_handle user,
+    _In_z_ const char* statName,
+    _In_z_ const char* statValue
+    ) XBL_NOEXCEPT;
+
+/// <summary> 
+/// Deletes a stat. Will clear stat from service and leaderboard information.
+/// </summary>
+/// <param name="user">The local user whose stats to access.</param>
+/// <param name="statName">The name of the statistic to delete.</param>
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: >tatName">The name of  [](start = 18, length = 21)

UTF-8 encoded

File: Include/xsapi-c/stats_manager_c.h [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+    /// <summary> 
+    /// UTF-8 encoded name of the stat
+    /// </summary>
+    _Field_z_ const char* name;
+    
+    /// <summary> 
+    /// The stat value as a double
+    /// </summary>
+    /// <remarks>Will be 0 if dataType is not XblStatsManagerStatDataType_Number</remarks>
+    double asNumber;
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: >asNumber [](start = 11, length = 8)

asDouble?

File: Include/xsapi-c/stats_manager_c.h [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+/// 
+/// Stats will be sent to the service automatically every 5 minutes or you can call 
+/// XblStatsManagerRequestFlushToService() if you want to update the service with 
+/// the latest stats sooner such as when a match or round ends.
+/// </summary>
+/// <param name="user">The local user whose stats to access.</param>
+/// <param name="statName">The name of the statistic to modify.</param>
+/// <param name="statValue">The value to replace the stat by.</param>
+/// <returns>HRESULT return code for this API operation.</returns>
+STDAPI XblStatsManagerSetStatAsNumber(
```
Comment (jasonsandlin) [inline Include/xsapi-c/stats_manager_c.h]: >Number( [](start = 31, length = 7)

AsDouble( ?

**Pull request closed without merging**